yt_dlp/extractor/twitter.py

```diff
@@ -816,6 +1049,138 @@ def _real_extract(self, url):
         return info
 
 
+class TwitterSpacesIE(TwitterBaseIE):
+    IE_NAME = 'twitter:spaces'
+    _VALID_URL = TwitterBaseIE._BASE_REGEX + r'i/spaces/(?P<id>[0-9a-zA-Z]{13})'
+    _TWITTER_GRAPHQL = 'https://twitter.com/i/api/graphql/HPEisOmj1epUNLCWTYhUWw/'
+
+    _TESTS = [{
+        'url': 'https://twitter.com/i/spaces/1RDxlgyvNXzJL',
+        'info_dict': {
+            'id': '1RDxlgyvNXzJL',
+            'ext': 'm4a',
+            'title': 'King Carlo e la mossa Kansas City per fare il Grande Centro',
+            'description': 'Twitter Space partecipated by annarita digiorgio, Signor Ernesto, Raffaello Colosimo, Simone M. Sepe',
+            'uploader': r're:Lucio Di Gaetano.*?',
+            'uploader_id': 'luciodigaetano',
+            'live_status': 'was_live',
+            'timestamp': 1659877956397,
+        },
+        'params': {
+            'skip_download': True,  # requires ffmpeg
+        },
+    }, {
+        # Twitter Space not started yet
+        'url': 'https://twitter.com/i/spaces/1mnGeRyljQPJX',
+        'info_dict': {
+            'id': '1mnGeRyljQPJX',
+            'title': '@staderlabs x Hedera AMA w/ @defigirlxoxo, @bmgentile, & @thehbarbull',
+            'description': 'Twitter Space partecipated by nobody yet.',
+            'uploader': 'Hedera',
+            'uploader_id': 'hedera',
+            'live_status': 'is_upcoming',
+            'timestamp': 1662394668924,
+        },
+        'params': {
+            'ignore_no_formats_error': True,
+            'skip_download': True,  # requires ffmpeg
+        },
+        'expected_warnings': ['Twitter Space not started yet', 'No video formats found', 'Requested format is not available'],
+    }]
+
+    def _parse_spaces_data(self, data, space_id):
```

Comment:
I think this should not be a seperate function. Pulling it back in to the extract function means we don't have to pop the `media_key` from the returned dict.